From f281b935985f1b592534ee2837ce0d0b28c7dc43 Mon Sep 17 00:00:00 2001
From: Andrey Zhizhikin <andrey.z@gmail.com>
Date: Tue, 11 Jan 2022 15:32:05 -0600
Subject: [PATCH] mkimage_fit_atf: fix fit generator node naming

Extend sections which contains "firmware" in their types with "os"
parameter to conform to the latest FIT specification.

Indicate that load and entry addresses are covered by one 32-bit value
by adding "address-cells" in the root node of the generated ITS.

Upstream-Status: Inappropriate [script is imx-boot specific]
Link: [1] https://github.com/u-boot/u-boot/commit/79af75f7776fc20b0d7eb6afe1e27c00fdb4b9b4
Signed-off-by: Andrey Zhizhikin <andrey.z@gmail.com>
Cc: Tom Hochstein <tom.hochstein@nxp.com>
---
 iMX8M/mkimage_fit_atf.sh | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/iMX8M/mkimage_fit_atf.sh b/iMX8M/mkimage_fit_atf.sh
index 4f2b3f3..10903ea 100755
--- a/iMX8M/mkimage_fit_atf.sh
+++ b/iMX8M/mkimage_fit_atf.sh
@@ -64,10 +64,12 @@ cat << __HEADER_EOF
 
 / {
 	description = "Configuration to load ATF before U-Boot";
+	#address-cells = <1>;
 
 	images {
 		uboot-1 {
 			description = "U-Boot (64-bit)";
+			os = "u-boot";
 			data = /incbin/("$BL33");
 			type = "standalone";
 			arch = "arm64";
@@ -93,6 +95,7 @@ done
 cat << __HEADER_EOF
 		atf-1 {
 			description = "ARM Trusted Firmware";
+			os = "arm-trusted-firmware";
 			data = /incbin/("$BL31");
 			type = "firmware";
 			arch = "arm64";
@@ -106,6 +109,7 @@ if [ -f $BL32 ]; then
 cat << __HEADER_EOF
 		tee-1 {
 			description = "TEE firmware";
+			os = "op-tee-firmware";
 			data = /incbin/("$BL32");
 			type = "firmware";
 			arch = "arm64";
-- 
2.17.1

